File: src/snippets/bot-protection/reference/remix/DenyingBots.ts
import arcjet, { detectBot } from "@arcjet/remix";
import type { LoaderFunctionArgs } from "@remix-run/node";

const aj = arcjet({
  key: process.env.ARCJET_KEY!,
  rules: [
    detectBot({
      mode: "LIVE",
      // configured with a list of bots to deny from
      // https://arcjet.com/bot-list - all other detected bots will be allowed
      deny: [
        "CATEGORY:AI", // denies all detected AI and LLM scrapers
        "CURL", // denies the default user-agent of the `curl` tool
      ],
    }),
  ],
});

export async function loader(args: LoaderFunctionArgs) {
  const decision = await aj.protect(args);

  if (decision.isDenied()) {
    throw new Response("Forbidden", { status: 403, statusText: "Forbidden" });
  }

  return null;
}

File: src/snippets/bot-protection/reference/python-flask/DenyingBots.py
import os

from arcjet import BotCategory, Mode, arcjet_sync, detect_bot

aj = arcjet_sync(
    key=os.environ["ARCJET_KEY"],
    rules=[
        detect_bot(
            mode=Mode.LIVE,
            # configured with a list of bots to deny from
            # https://arcjet.com/bot-list — all other detected bots will be allowed
            deny=[
                BotCategory.AI,  # denies all detected AI and LLM scrapers
                "CURL",  # denies the default user-agent of the `curl` tool
            ],
        ),
    ],
)
